config.yaml

```diff
@@ -41,7 +41,12 @@ params:
     db: test_data/mp2_test_db/mpa_v20_m200
   kraken:
     db: test_data/kraken_test_db
+    # Bracken kmer distribution file
     kmers: test_data/kraken_test_db/75mers_distr.txt
+    # taxonomic ranks (kingdom to species) to reestimate abundance using Bracken
+    levels: phylum,genus,species
```

Comment:
why those levels? Do we really need to store phylum and genus if we keep species? Once we know the taxonomy lineages, we should be able to recover higher ranks, right?

Reply:
These three levels are the most popular results. Re. your question, NO. We can recover higher ranks from lower ranks. Not every taxon has all 8 ranks defined. There are lots of say "unidentified Bacteroides".

Reply:
This is a recurring and interesting problem. 

A fundamental difference between amplicon marker gene profile and general WGS profile is that for amplicon you can always count the # of units you have, even if you can't assign a confident taxon to it. In WGS you might be able to assign a read to a position on a phylogeny or taxonomy, but not say whether it is same or different from a different read to a different region that was assigned to the same rank. 

Most of our tools only deal well with tips. Ultimately I don't think they need to, though. Unifrac, for example, can conceptually work with #s assigned to internal nodes on a tree as well as tips. 

For now I think that it's good to keep these separate tables because that is how Bracken outputs, and Qiyun's point is well taken re: common demand for table types. But would be nice to keep other options in back of mind for future.

Reply:
Jon thanks for your comments! Makes sense! With consideration of this I kept the original Kraken outputs. Let's see how Ben and Gabe engineer their tool to make it not dependent on taxonomic ranks. I guess that would be challenging.